### PR TITLE
Service Development Guide

### PR DESCRIPTION
**Pull request by CatGarab**

### CODE REVIEW

**Comment by CatGarab** [comment]
[Service Dev Guide](https://trello.com/c/6H8kRJug/893-service-dev-guide)

File: docs/os-docs/services/service-dev.rst [hunk 1]
```diff
@@ -1,8 +1,180 @@
...
+
+Logging
+-------
+
+We recommend logging any errors which your service encounters.
+These errors will likely also be returned in a GraphQL response, however we feel that this
+redudancy is important for maintaining overall system health.
+
+All log messages issued by the service should be routed to the system logs.
+You may also choose to echo the messages to ``stdout``, however that is not a required behavior.
```
Comment (cantino) [inline docs/os-docs/services/service-dev.rst]: Where does stdout go for a running service?

File: docs/os-docs/services/service-dev.rst [hunk 2]
```diff
@@ -1,8 +1,180 @@
...
+
+By default, all services require that the IP address and port of their GraphQL endpoint be defined
+in the system's `config.toml` file.
+It is generally assumed that any port which is not already listed in the config file is available
+for use.
+
+It may be useful for your service to have certain additional configurable settings.
+For instance, you might want to be able to adjust certain timeout values, or change which device
+bus your hardware is connected to.
+In this case, your service should read the needed configuration values from the system's
```
Comment (cantino) [inline docs/os-docs/services/service-dev.rst]: Newbie question: so all services share a single config file?

File: docs/os-docs/services/service-dev.rst [hunk 3]
```diff
@@ -1,8 +1,180 @@
...
+Unit and integration tests are a good way to ensure that your service remains functional and
+compatible with the KubOS ecosystem over time.
+More information about setting up testing can be found in our :doc:`testing <../../contributing/testing>`
+doc.
+
+Creating an Init Script
+-----------------------
+
+If you would like your service to be automatically started at system boot, you will need to create
+an init script.
```
Comment (cantino) [inline docs/os-docs/services/service-dev.rst]: Maybe link to a general guide about Unix init scripts?

File: docs/os-docs/services/service-dev.rst [hunk 4]
```diff
@@ -1,8 +1,180 @@
...
+following format: ``S{run-level}{application-name}``. The run-level value should be between 1 and 99.
+The lower the value, the earlier it will be run in the system boot process.
+
+Installing Your Service
+-----------------------
+
+Once you have finished service development, you should install the service in its final location in
+your OBC.
+
+Custom services may either live in the user data partition or in the root file system.
```
Comment (cantino) [inline docs/os-docs/services/service-dev.rst]: Maybe a sentence on why one would choose one vs the other?

File: docs/os-docs/services/service-dev.rst (hunk 2 above)
Comment (jacoffey3) [reply docs/os-docs/services/service-dev.rst]: Indeed. It's also what the apps use to see how/where to access services

File: docs/os-docs/services/service-dev.rst (hunk 3 above)
Comment (jacoffey3) [reply docs/os-docs/services/service-dev.rst]: Seconded

File: docs/os-docs/services/service-dev.rst (hunk 4 above)
Comment (jacoffey3) [reply docs/os-docs/services/service-dev.rst]: "If the service interacts with core avionics or communications hardware, and is not expected to change after launch, we recommend including it in the root file system for recovery purposes. Only services included in the root file system are recovered automatically. Check out the documentation on the recovery process <link> for information on how it relates to the root file system."

File: docs/os-docs/services/service-dev.rst (hunk 3 above)
Comment (CatGarab) [reply docs/os-docs/services/service-dev.rst]: I looked, but I couldn't find a good one :/

File: docs/os-docs/services/service-dev.rst (hunk 1 above)
Comment (plauche) [reply docs/os-docs/services/service-dev.rst]: The great void :rocket:

**Pull request merged**